### PR TITLE
Update Typescript to 4.0.5

### PR DESCRIPTION
**Pull request by filipenevola**
Based on #11154

### CODE REVIEW

File: tools/fs/files.ts [hunk 1]
```diff
@@ -373,12 +373,12 @@ export const blankHash = "47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=";
 // Returns a base64 SHA256 hash representing a tree on disk. It is not sensitive
 // to modtime, uid/gid, or any permissions bits other than the current-user-exec
 // bit on normal files.
-export function treeHash(root: string, options: {
-  ignore: (path: string) => boolean;
+export function treeHash(root: string, optionsParams: {
```
Comment (filipenevola) [inline tools/fs/files.ts:376]: @afrokick can you review my change? I'm not a TS expert but I did the change based on my general knowledge ;) It was breaking with TS2783 error

Comment (afrokick) [reply tools/fs/files.ts:376]: LGTM

File: scripts/dev-bundle-tool-package.js [hunk 2]
```diff
@@ -14,8 +14,8 @@ var packageJson = {
     pacote: "https://github.com/meteor/pacote/tarball/a81b0324686e85d22c7688c47629d4009000e8b8",
     "node-gyp": "6.0.1",
     "node-pre-gyp": "0.14.0",
-    typescript: "3.8.3",
-    "meteor-babel": "7.9.0",
+    typescript: "4.0.5",
```
Comment (afrokick) [inline scripts/dev-bundle-tool-package.js:17]: also we need to update meteor-babel's TS to 4.0.5

Comment (filipenevola) [reply scripts/dev-bundle-tool-package.js:17]: done

**Comment by filipenevola** [comment]
Windows tests are getting stuck, not sure why. I'll merge it and so we can see the tests in the release-1.12 branch.

**Pull request merged**